core/src/main/resources/hudson/PluginManager/installed.jelly

```diff
@@ -162,7 +162,18 @@ THE SOFTWARE.
                     </j:if>
                   </td>
                   <j:set var="state" value="${p.enabled?'true':null}"/>
-                  <td class="jenkins-table__cell--tight enable" data="${state}">
+<!--
+Sort order (only '1' / '2-3' / '4-7' is trivially reflected on the UI through the state/existence of the toggle button):
+1. Plugins marked for deletion (temporary until restart)
+2. Inactive and disabled plugins (often present for a while before uninstalled)
+3. Active plugins that have been disabled (temporary until restart)
+4. Inactive plugins that have just been enabled (temporary until restart)
+5. Enabled and active plugins that can be freely disabled/uninstalled from a dependency POV
+6. Enabled and active plugins with implied dependants, can be disabled but might be unsafe
+7. Enabled and active plugins with mandatory dependants, cannot be disabled.
+This groups by visual presentation, with the states that can be more readily interacted with towards the top, pushing required dependencies to the bottom.
+-->
+                  <td class="jenkins-table__cell--tight enable" data="${p.isDeleted() ? '1' : (!p.enabled ? (p.active ? '3' : '2') : (p.active ? (p.hasMandatoryDependents() ? '7' : (p.hasImpliedDependents() ? '6' : '5')) : '4'))}">
```

Comment:
I would extract this to var with normal jelly when conditions.
1. it will improve readability
2. this value if needed can be used elsewhere.

Reply:
Jelly `case`/`when`/`otherwise` are a verbose mess, so I'd rather not (3 tags/6 lines per condition plus 7 assignments makes ~36-43 lines for this incomplete version I still want to amend). YAGNI for item 2; can always be changed once there's a need.

Reply:
In my plugins I expose those complicated expressions in the Java object model, where I can write a unit test as well. Using too much logic in jelly is not recommended.

Reply:
@uhafner this is even better idea :)